stuff/canonize_tags.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/python3
+
+'''Ensures the usage of just canonical tags on quality nominations.
+
+This script reads all quality nominations suggestions contents
+and ensures that only canonical tags are used.
+'''
+
+import argparse
+import json
+import logging
+import os
+import unicodedata
+import re
+from typing import Mapping, Set
+
+import MySQLdb.constants.ER
+
+import cron.lib.db as db
+import cron.lib.logs as logs
+
+LOCALIZATIONS_ROOT = os.path.abspath(os.path.join(__file__,
+                                                  '..', '..',
+                                                  'frontend/www/js/omegaup'))
+LANGS = ['en', 'es', 'pt']
+
+
+def normalize_tag(tag: str) -> str:
+    '''Normalizes tags, similar to Tags::normalize() in PHP'''
+    # Remove empty spaces
+    tag = tag.strip()
+    # Remove accents
+    tag = unicodedata.normalize('NFD',
+                                tag).encode('ascii', 'ignore').decode("utf-8")
+    # Use '-' for splitting if necessary
+    tag = re.sub(r'[^a-z0-9]', '-', tag.lower())
+    tag = re.sub(r'--+', '-', tag)
+    return tag
+
+
+def insert_new_tags(tags: Set[str],
+                    cur: MySQLdb.cursors.DictCursor) -> None:
+    '''Inserts new problem tags inside Tags table on DB'''
+    logging.info('Inserting new Tags on database')
+    cur.executemany('''INSERT IGNORE INTO `Tags`(`name`)
+                    VALUES (%s);''',
+                    [(tag,) for tag in tags])
+
+
+def get_inverse_mapper(
+        cur: MySQLdb.cursors.DictCursor) -> Mapping[str, Mapping[str, str]]:
+    '''Gets the inverse mapper for problem tags entries in lang files'''
+    logging.info('Getting tags mapper from lang files.')
+    new_tags = set()
+    inverse_mapper = {}
+    for lang in LANGS:
+        path = os.path.join(LOCALIZATIONS_ROOT, 'lang.%s.json' % (lang))
+        with open(path, 'r') as f:
+            mappings = json.load(f)
+            inverse_mapping = {}
+            for key, value in mappings.items():
+                if key.startswith('problemTopic'):
```

Comment:
nit: aquí puedes usar el guard clause pattern:

```suggestion
                if not key.startswith('problemTopic'):
                    continue
                # el resto del ciclo.
```